### DISCUSSION TITLE
How to find which code runs on the global thread pool

### DISCUSSION THREAD

**Discussion by alexpyattaev** [rayon-rs/rayon#1218] [Q&A]
I am working on a large codebase and I am trying to find which of the code relies on the global thread pool. Is there any reasonable way to do so?  Maybe there is a way to disable it entirely?

**Comment by cuviper** [comment]
Maybe in a debugger you could put a breakpoint here:
https://github.com/rayon-rs/rayon/blob/d179ea9fa69cab08a3b2762ff9b15249a71b9b94/rayon-core/src/registry.rs#L164

**Reply by alexpyattaev** [reply]
Yeah, that is one way to do it, though it is a mess to set up every time in a complex build. Thanks for the advice!
